server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -122,30 +115,50 @@ public TemporalAccessor parse(String input) {
         }
 
         try {
-            return parser.parse(input);
+            return doParse(input);
         } catch (DateTimeParseException e) {
             throw new IllegalArgumentException("failed to parse date field [" + input + "] with format [" + format + "]", e);
         }
     }
 
+    private TemporalAccessor doParse(String input) {
+        if (parsers.size() > 1) {
+            for (DateTimeFormatter formatter : parsers) {
+                ParsePosition pos = new ParsePosition(0);
+                Object object = formatter.toFormat().parseObject(input, pos);
+                if (parsingSucceeded(object, input, pos) == true) {
+                    return (TemporalAccessor) object;
+                }
+            }
+            throw new DateTimeParseException("Failed to parse with all enclosed parsers", input, 0);
+        }
+        return firstParser().parse(input);
+    }
+
+    private boolean parsingSucceeded(Object object, String input, ParsePosition pos) {
```

Comment:
this is only used once. it might be more readable to just have a boolean

```
boolean isParsingSuccessful = object != null && pos.getIndex() == input.length();
```

in the other method?

Reply:
it is used once, but my intention was to hide the implementation details and this expression in the other method